LayoutKit/Layouts/TextViewLayout.swift

```diff
@@ -0,0 +1,177 @@
+                  viewReuseId: viewReuseId,
+                  configure: configure)
+    }
+
+    // MARK: - Layout protocol
+
+    open func measurement(within maxSize: CGSize) -> LayoutMeasurement {
+        let fittedSize = textSize(within: maxSize)
+        return LayoutMeasurement(layout: self, size: fittedSize.decreasedToSize(maxSize), maxSize: maxSize, sublayouts: [])
+    }
+
+    open func arrangement(within rect: CGRect, measurement: LayoutMeasurement) -> LayoutArrangement {
+        let frame = alignment.position(size: measurement.size, in: rect)
+        return LayoutArrangement(layout: self, frame: frame, sublayouts: [])
+    }
+
+    // MARK: - private helpers
+
+    private func textSize(within maxSize: CGSize) -> CGSize {
+        let options: NSStringDrawingOptions = [
+            .usesLineFragmentOrigin
+        ]
+
+        var size: CGSize
+        let heightInset = (textContainerInset.top + textContainerInset.bottom) + (layoutMargins.top + layoutMargins.bottom)
+        let widthInset = (textContainerInset.left + textContainerInset.right) + (layoutMargins.left + layoutMargins.right) + lineFragmentPadding
+            + (contentInset.left + contentInset.right)
+        let textContainerMaxSize = CGSize(width: maxSize.width + widthInset, height: maxSize.height + heightInset)
+        switch text {
+        case .attributed(let attributedText):
+            if attributedText.length == 0 {
+                return .zero
+            }
+
+            // UITextView uses a default font if one is not specified in the attributed string.
+            // boundingRectWithSize does not appear to have the same logic,
+            // so we need to ensure that our attributed string has a default font.
+            // We do this by creating a new attributed string with the default font and then
+            // applying all of the attributes from the provided attributed string.
+            let fontAttribute = [NSFontAttributeName: font]
+            let attributedTextWithFont = NSMutableAttributedString(string: attributedText.string, attributes: fontAttribute)
+            let fullRange = NSMakeRange(0, (attributedText.string as NSString).length)
+            attributedTextWithFont.beginEditing()
+            attributedText.enumerateAttributes(in: fullRange, options: .longestEffectiveRangeNotRequired, using: { (attributes, range, _) in
+                attributedTextWithFont.addAttributes(attributes, range: range)
+            })
+            attributedTextWithFont.endEditing()
+
+            size = attributedTextWithFont.boundingRect(with: textContainerMaxSize, options: options, context: nil).size
+        case .unattributed(let text):
+            if text == "" {
+                return .zero
+            }
+            size = text.boundingRect(with: textContainerMaxSize, options: options, attributes: [NSFontAttributeName: font], context: nil).size
+        }
+
+        size.height = ceil(size.height + heightInset)
```

Comment:
The LabelLayout uses roundedUpToFractionalPoint instead of ceil. Which one is appropriate here and why?